src/app/domain/survey.py

```diff
@@ -0,0 +1,191 @@
+        to_number = page_num * QUESTIONS_PER_PAGE
+
+        return from_number, to_number
+
+    def _normalize_question_numbers(self, questions, page_num, out=True):
+        """
+        Convert short question numbers to regular numbers, or vice versa.
+        :param questions:
+        :param page_num:
+        :param out:
+        :return:
+        """
+        return questions
+
+
+class ShortASurvey(Survey):
+    QUESTIONS = [
+        2, 5, 8, 10, 13, 20, 22, 23, 25, 28, 30, 31, 32, 34, 35, 38, 40, 43, 45, 48, 50, 52,
+        53, 54, 56, 57, 60, 61, 62, 63, 64, 65, 68, 70, 71, 74, 75, 76, 77, 79, 80, 82, 83,
+        86, 88, 89, 91, 96, 97, 98, 101, 104, 106, 110, 112, 114, 118, 120, 123, 126, 127,
+        130, 132, 134, 136, 137, 138, 139, 140, 142, 143, 146, 147, 148, 149, 152, 153, 155,
+        157, 160, 161, 163, 166, 167, 169, 171, 174, 176, 178, 180
+    ]
+
+    def __init__(self, user_id, quiz_type='short_a'):
+        """
+        :param user_id:
+        :return:
+        """
+        max_pages = len(self.QUESTIONS) / QUESTIONS_PER_PAGE
+        super(ShortASurvey, self).__init__(user_id, quiz_type=quiz_type, max_pages=max_pages)
+
+    def _normalize_question_numbers(self, questions, page_num, out=True):
+        """
+        Convert short question numbers to regular numbers, or vice versa.
+        :param questions:
+        :param page_num:
+        :param out:
+        :return:
+        """
+        if len(questions) == 0:
+            return questions
+
+        from_num, to_num = self._calculate_from_and_to_for_page_number(page_num)
+        index = 0
+        for new_question_number in range(from_num, to_num + 1):
+            if not out:
+                # The questions come in as QuizAttemptAnswers so we have to access their attribute.
+                questions[index].question_number = self.QUESTIONS[new_question_number - 1]
+            else:
+                # The questions go out as regular dictionaries to we have to set the index.
+                questions[index]['question_number'] = new_question_number
+            index += 1
+
+        return questions
+
+    def _get_questions_for_survey_page(self, page_num):
+        from_num, to_num = self._calculate_from_and_to_for_page_number(page_num)
+
+        questions = self.QUESTIONS[from_num - 1:to_num]
+        return [Question.get_by_question_number(q_num) for q_num in questions]
+
+
+class ShortBSurvey(ShortASurvey):
+    QUESTIONS = [
+        1, 3, 4, 6, 7, 9, 11, 12, 14, 15, 16, 17, 18, 19, 21, 24, 26, 27, 29, 33, 36, 37,
+        39, 41, 42, 44, 46, 47, 49, 51, 55, 58, 59, 66, 67, 69, 72, 73, 78, 81, 84, 85, 87,
+        90, 92, 93, 94, 95, 99, 100, 102, 103, 105, 107, 108, 109, 111, 113, 115, 116, 117,
+        119, 121, 122, 124, 125, 128, 129, 131, 133, 135, 141, 144, 145, 150, 151, 154, 156,
+        158, 159, 162, 164, 165, 168, 170, 172, 173, 175, 177, 179
+    ]
+
+    def __init__(self, user_id, quiz_type='short_b'):
```

Comment:
The effective amount of code required to add new/different lengths of surveys is now quite small. Just a list of question numbers and a different "constructor" function (that isn't really what it is but it's close enough)